.github/workflows/ci.yml

```diff
@@ -0,0 +1,48 @@
+name: Continuous Integration
+on:
+  # Trigger the workflow on push or pull request, only for the main branch
+  push:
+    branches:
+      - main
+  pull_request:
+    branches:
+      - main
+
+jobs:
+  build:
+    name: Full Build (node ${{ matrix.node_version }})
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        node_version:
+          - 18.x
+      # https://stackoverflow.com/questions/61070925/github-actions-disable-auto-cancel-when-job-fails
+      fail-fast: false
+
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v2
+
+      - name: Set up Node.js
+        uses: actions/setup-node@v4
+        with:
+          node-version: ${{ matrix.node_version }}
+          cache: 'yarn'
+
+      - name: Install dependencies
+        run: yarn
+
+      - name: Build
+        run: yarn run ci
+
+      - name: Coveralls
+        uses: coverallsapp/github-action@1.1.3
+        with:
+          github-token: ${{ secrets.GITHUB_TOKEN }}
```

Comment:
consider removing this step -> we handle code coverage ourselves (merging package coverage reports, etc.) via the `ci` process.